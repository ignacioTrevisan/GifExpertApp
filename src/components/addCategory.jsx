import { useState } from "react"
import PropTypes from 'prop-types'
export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }
    const onSubmit = (event) => {
        if (inputValue.length == 0) return;
        event.preventDefault();
        setInputValue("");
        onNewCategory(inputValue);
    }
    return <form onSubmit={onSubmit} aria-label="form">
        <input
            type="text"
            placeholder="Buscar gifs"
            value={inputValue}
            onChange={onInputChange}
        />
    </form>

}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}