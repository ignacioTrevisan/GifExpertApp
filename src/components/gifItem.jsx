import PropTypes from 'prop-types'

export const GifItem = ({ img, Titulo, id }) => {
    return (
        <div className="card">
            <img src={img} alt={Titulo}></img>
            <p>{Titulo}</p>
        </div>
    )
}
GifItem.propTypes = {
    Titulo: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
}