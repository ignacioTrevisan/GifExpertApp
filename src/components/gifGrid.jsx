
import { GifItem } from './gifItem';
import { UseFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

    const { images, isLoading } = UseFetchGifs(category);

    return (
        <>

            <h1>{category}</h1>
            {isLoading && <h2>Cargando...</h2>}
            <div className='card-grid'>
                {images.map(img => <GifItem {...img} key={img.id} />)}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
