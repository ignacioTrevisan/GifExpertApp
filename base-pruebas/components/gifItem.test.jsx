
/**

 * @jest-environment jsdom

 */
import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components';

describe('Prueba a <GifItem />', () => {
    test('Hacer match con el spanshot', () => {
        const imagen = "https://media4.giphy.com/media/UPYCIVX3ABQoLAGO7W/giphy.gif?cid=ffddb294jme2owkru490riardwfc23ijeoy1edebeartqe7b&ep=v1_gifs_search&rid=giphy.gif&ct=g";
        const { container } = render(<GifItem Titulo="Tu hermana" img={imagen} />)
        expect(container).toMatchSnapshot();
        screen.debug();
    })
    test('Hacer mostrar el titulo indicado y el alt indicado', () => {
        const imagen = "https://media4.giphy.com/media/UPYCIVX3ABQoLAGO7W/giphy.gif?cid=ffddb294jme2owkru490riardwfc23ijeoy1edebeartqe7b&ep=v1_gifs_search&rid=giphy.gif&ct=g";
        const Titulo = "Tu hermana";
        render(<GifItem Titulo={Titulo} img={imagen} />)
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(imagen);
        expect(alt).toBe("Tu hermana");

    })

})