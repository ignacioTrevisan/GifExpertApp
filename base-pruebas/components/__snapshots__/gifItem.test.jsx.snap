// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba a <GifItem /> Hacer match con el spanshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Tu hermana"
      src="https://media4.giphy.com/media/UPYCIVX3ABQoLAGO7W/giphy.gif?cid=ffddb294jme2owkru490riardwfc23ijeoy1edebeartqe7b&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Tu hermana
    </p>
  </div>
</div>
`;
